Update EIP-3540: Claify, improve and fill missing sections of EOF EIPs (#9662)
ethereum/EIPs: EIPS/eip-4200.md

## Security Considerations

> Adding new instructions with immediate arguments needs to be carefully considered when implementing EOF container validation algorithm.

Reviewer: for clarity 

```suggestion
Adding new instructions with immediate arguments should be carefully considered when implementing the EOF container validation algorithm.
```

> Static relative jumps execution does not require runtime check of the jump destination. It greatly reduces execution cost. Therefor the new opcodes gas cost can also be significantly reduced.

Reviewer: Therefore*
